src/EntryPoint.sol

```diff
@@ -700,9 +672,18 @@ contract EntryPoint is
                     revert(mload(0x40), returndatasize())
                 }
                 if iszero(mload(0x00)) { mstore(0x00, shl(224, 0x6c9d47e8)) } // `CallError()`.
-                return(0x00, 0x20) // Return the `err`.
+                revert(0x00, 0x20) // Revert with the `err`.
             }
-            return(0x60, 0x20) // If all success, returns with zero `err`.
+        }
+
+        uint256 remainingPaymentAmount = u.totalPaymentAmount - u.prePaymentAmount;
```

Comment:
Can use `Math.rawSub(u.totalPaymentAmount, u.prePaymentAmount)`, or unchecked math.

The check on 
```solidity
u.prePaymentMaxAmount > u.totalPaymentMaxAmount
     || u.prePaymentAmount > u.prePaymentMaxAmount
     || u.totalPaymentAmount > u.totalPaymentMaxAmount
```

makes it safe.